doc/languages-frameworks/beam.section.md

```diff
@@ -0,0 +1,116 @@
+# BEAM Languages (Erlang, Elixir & LFE)
+
+## Introduction
+
+In this document and related Nix expressions, we use the term, *BEAM*,
+to describe the environment. BEAM is the name of the Erlang Virtual
+Machine and, as far as we're concerned, from a packaging perspective,
+all languages that run on the BEAM are interchangeable. That which
+varies, like the build system, is transparent to users of any given BEAM
+package, so we make no distinction.
+
+## Structure
+
+All BEAM-related expressions are available via the top-level `beam`
+attribute, which includes:
+
+  - `interpreters`: a set of compilers running on the BEAM, including
+    multiple Erlang/OTP versions (`beam.interpreters.erlangR19`, etc),
+    Elixir (`beam.interpreters.elixir`) and LFE
+    (`beam.interpreters.lfe`).
+
+  - `packages`: a set of package builders (Mix and rebar3), each
+    compiled with a specific Erlang/OTP version, e.g.
+    `beam.packages.erlangR19`.
+
+The default Erlang compiler, defined by `beam.interpreters.erlang`, is
+aliased as `erlang`. The default BEAM package set is defined by
+`beam.packages.erlang` and aliased at the top level as `beamPackages`.
+
+To create a package builder built with a custom Erlang version, use the
+lambda, `beam.packagesWith`, which accepts an Erlang/OTP derivation and
+produces a package builder similar to `beam.packages.erlang`.
+
+Many Erlang/OTP distributions available in `beam.interpreters` have
+versions with ODBC and/or Java enabled or without wx (no observer
+support). For example, there's `beam.interpreters.erlangR22_odbc_javac`,
+which corresponds to `beam.interpreters.erlangR22` and
+`beam.interpreters.erlangR22_nox`, which corresponds to
+`beam.interpreters.erlangR22`.
+
+## Build Tools
+
+### Rebar3
+
+We provide a version of Rebar3, under `rebar3`. We also provide a helper
+to fetch Rebar3 dependencies from a lockfile under `fetchRebar3Deps`.
+
+### Mix & Erlang.mk
+
+Both Mix and Erlang.mk work exactly as expected. There is a bootstrap
+process that needs to be run for both, however, which is supported by
+the `buildMix` and `buildErlangMk` derivations, respectively.
+
+## How to Install BEAM Packages
+
+BEAM builders are not registered at the top level, simply because they
+are not relevant to the vast majority of Nix users. To install any of
+those builders into your profile, refer to them by their attribute path
+`beamPackages.rebar3`:
+
+    $ nix-env -f "<nixpkgs>" -iA beamPackages.rebar3
```

Comment:
Please use triple backticks for code with `ShellSession` language.

Reply:
Unfortunately, this is a regression in terms of usability (`$` is now unmarked so it will be selected when a person double clicks the line, making copying very annoying) but unless we keep `screen` tags (they are actually not so much noise compared to markdown so maybe that would be fine), it is not really expressible in markdown.

Reply:
Ah I didn't notice this. The old way with docbook was definitely better in terms of usability. Will keep as is anyway, unless folks in general want to keep the `screen` tags for stuff like this.